openshiftcontrolplane/v1/types.go

```diff
@@ -46,6 +46,13 @@ type OpenShiftAPIServerConfig struct {
 
 	// TODO this needs to be removed.
 	APIServerArguments map[string][]string `json:"apiServerArguments"`
+
+	// apiServers is a list of API servers to enable.
+	// - '*' enables all on-by-default API servers
```

Comment:
Magic value strikes me as potentially problematic, why not make a structure that makes this more explicit? Is the intention to allow people to combine `*` with disabled by default api servers? What happens if I use `*` and then include `-<default-api-server>`? 

Is there a pattern for this in upstream that we might be able to take inspiration from?

Reply:
I would probably lean into creating a structured API rather than using a string based/wildcard based API like that. Just because upstream does it that way doesn't mean we can't do something that is more structured but still convertible.

If you had two slices, `Enabled` and `Disabled` where an empty `Enabled`, defaulted (probably internally rather than in the API) to `*` being passed into that upstream function, you could easily reconstruct the input you need, but have a better, structured API downstream

I assume as well you'd want to make it so that one controller cannot be specified in both Enabled and Disabled lists, you can do that with CEL for CRDs, a touch harder for openshift-apiserver but by no means impossible

Reply:
Updated based on your suggestion. The validation of the fields will be performed during the OA operator initialization.